src/landIce/evaluators/LandIce_ResponseSurfaceVelocityMismatch_Def.hpp

```diff
@@ -239,21 +255,23 @@ void LandIce::ResponseSurfaceVelocityMismatch<EvalT, Traits>::evaluateFields(typ
 
             t += sum * w_measure(sideSet_idx,qp);
           }
-          this->local_response_eval(cell, 0) += t*scaling*alpha;//*50.0;
-          this->global_response_eval(0) += t*scaling*alpha;//*50.0;
-          p_reg += t*scaling*alpha;
-        }
+
+          ScalarT result = t*scaling*alpha;
+          KU::atomic_add<ExecutionSpace>(&(this->local_response_eval(cell, 0)), result);//*50.0
+          KU::atomic_add<ExecutionSpace>(&(this->global_response_eval(0)), result);//*50.0
```

Comment:
Would it make sense to move the update of the global resp and p_reg to outside the kernel? Like
```c++
// kernel to update local_resp(cell,0)
parellel_reduce ( ...); // ||reduce to reduce local resp into global resp
deep_copy(p_ref, g_resp);
```
I'm not sure if atomic adds are going to be noticeable though. If not, then maybe it's ok to leave as is.

Reply:
Yeah, this kernel is pretty tiny in terms of overall runtime (and is faster than what we have now) so I figured it's probably fine.